netsim-ffi/src/lib.rs

```diff
@@ -86,8 +87,20 @@ pub unsafe extern "C" fn netsim_context_open(
     if context.is_null() || output.is_null() {
         SimError::NullPointerArgument
     } else {
-        // TODO
+        let mut context = Box::from_raw(context);
+        match context.open() {
+            Ok(sim_socket) => {
+                **output = sim_socket;
```

Comment:
This will not work, you need to allocate the sim_socket on the heap otherwise the content will be released from memory.

```rust
     *output = Box::into_raw(Box::new(sim_socket));
```